src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1116,6 +1108,80 @@ bool FullyConnected::tryExtractParamForLLMFc(llmdnn::fc_create_param& param) {
     return false;
 }
 
+MemoryPtr FullyConnected::castMemoryPtr(MemoryPtr memPtr, const InferenceEngine::Precision prec) {
+    auto engine = memPtr->getEngine();
+    dnnl::stream engine_stream(engine);
+
+    // get weightPtr and dnnlweightDesc
+    auto dnnlMemDesc = memPtr->GetDescWithType<DnnlMemoryDesc>()->getDnnlDesc();
+    // desc_ptr
+    auto newMemDescPtr = memPtr->getDescPtr()->cloneWithNewPrecision(prec);
+    // dnnl_desc_ptr
+    DnnlMemoryDescPtr dnnlNewMemPtr = MemoryDescUtils::convertToDnnlMemoryDesc(newMemDescPtr);
+    // dnnl_desc
+    auto dnnlNewMemDesc = dnnlNewMemPtr->getDnnlDesc();
+    MemoryPtr newMemPtr = MemoryPtr(new Memory(engine));
+    newMemPtr->Create(dnnlNewMemPtr);
+
+    auto reorder = getReorderPrim(context->getParamsCache(), engine, dnnlMemDesc, dnnlNewMemDesc);
+    primArgs = {{DNNL_ARG_SRC, memPtr->GetPrimitive()}, {DNNL_ARG_DST, newMemPtr->GetPrimitive()}};
+    reorder.execute(engine_stream, primArgs);
+    engine_stream.wait();
+    return newMemPtr;
+}
+
+void FullyConnected::tryCompressWeightForLLMFc() {
+    const auto inPrec = getOriginalInputPrecisionAtPort(DATA_ID);
+    bool needCompress = dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx) &&
+        one_of(inPrec, InferenceEngine::Precision::BF16);
+    // weight
+    // If the input precision is fp32 on SPR, the weight will be cast to bf16 for acceleration. 
+    auto weightPtr = needCompress ?
```

Comment:
The callee has checked the getOriginalInputPrecisionAtPort(DATA_ID) should be bf16, needCompress should be removed.

Reply:
removed